src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -296,12 +325,18 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
         {
         }
 
-        void load(const MergeTreeData & data, const DiskPtr & disk_, const String & part_path);
-        void store(const MergeTreeData & data, const DiskPtr & disk_, const String & part_path, Checksums & checksums) const;
+#if USE_ROCKSDB
+        void load(const MergeTreeData & data, const PartMetadataCachePtr & metadata_cache, const DiskPtr & disk, const String & part_path);
+#else
+        void load(const MergeTreeData & data, const DiskPtr & disk, const String & part_path);
```

Comment:
Lots of preprocessor macros in pretty generic code, maybe the interface should be simplified?

Reply:
Sorry, I can't think of a simpler implementation, what do you advide ?

Reply:
Maybe it worth introducing something like IPartMetadataLoader, that will have two implementations with rocksdb cache and w/o? Or different MinMaxIndex implementations. Although maybe complexity does not worth it (but too much ifdefs looks not great)

P.S. BTW the argument name is different in the declaration and implementation parts.

Reply:
done. See IPartMetadataManager